cardano-node/src/Cardano/Node/Handlers/TopLevel.hs

```diff
@@ -91,6 +93,17 @@ toplevelExceptionHandler prog = do
     rethrowExitCode :: ExitCode -> IO a
     rethrowExitCode = throwIO
 
+    rethrowDiffusionErrorExceptionInLinkedThreadExitSuccess :: Network.Failure -> IO a
+    rethrowDiffusionErrorExceptionInLinkedThreadExitSuccess full =
+      case full of
+        Network.DiffusionError e ->
+          case fromException (toException e) of
+            Just (ExceptionInLinkedThread _ eInner)
+              | Just ExitSuccess <- fromException eInner
+              -> throwIO ExitSuccess
+            _ -> throwIO full
+        _ -> throwIO full
```

Comment:
This is the fix.

Reply:
@coot I see that your suggestion is now outdated. Are you okay with the above solution?